### DISCUSSION TITLE
Project: The Download Analysis Robot

### DISCUSSION THREAD

**Discussion by mikekasprzak** [ludumdare/ludumdare#2028] [Ideas]
This is something for the future: Create a robot that looks-at and sends reports on the health of download links.

This can be a general process that's always run, and over time gives a health report for all links, even old ones.

* Are they even valid?
* Does the Windows build include an exe file?
* If Unity, is the `data` folder included?

**Comment by Cerno-b** [comment]
- If Gamemaker, is it an installer (if that's possible to find out)?
- If itch.io, was Windows download marked as "Windows" (otherwise the app will not install the game)

**Comment by local-minimum** [comment]
The `itch.io` I really would like too.

There's a bit of complexity here though.
Some robots only need to check linked page HTML to figure out stuff is correctly marked.
Some other robots would need to navigate the page to other pages to verify that everything needed is included.

It sounds like fun work to write those!

**Comment by zwrawr** [comment]
I don't think this needs to be a bot per-say, it could be done user side at the point of publishing.

**Comment by local-minimum** [comment]
I was about to add that this __has__ to be done user side because the need for this will be high during the last hours of compo/jam and that is when the server will be straining anyhow.

What we could potentially do on a per upload site basis do some sort of hash or similar of the actual download and add that to what is submitting. That way we can actually know with some fidelity when people have updated their game after submission and assist them in indicating that they need to clarifying the changes they made etc.

**Comment by Jezzamonn** [comment]
If it was server side this could get bandwidth heavy pretty quickly 😝  Potentially it could be rate limited to just scan a few games at a time, at the cost of taking a while to scan all the games.

Still, user side is a little limited too. If we want to check that the unity data folder is there, you have to download their game, right? Which for a large game, is kinda weird. Making someone wait for their game to upload AND download to submit a game isn't a great user experience. We can do some basic checks though. 

This problem would really be solved (or heavily mitigated) with a notification feature. What happens now is people comment saying "Hey, I couldn't play your game", but game authors don't see it. I usually send a tweet to someone when I find something like this and people respond quickly.

**Comment by local-minimum** [comment]
@Jezzamon, I think a report functionality that is coupled to mail notification and a smaller check on server side is the way to go. Checks client side isn't really possible because we're not uploading to our site just asking for a link.

I started working on some basic checks here

https://github.com/local-minimum/ludumdare/tree/link-checker?files=1
